### ISSUE TITLE
Error when trying to use decco for variants

### ISSUE THREAD

**Issue by mrmurphy** [ryb73/ppx_decco#28]
Very strange, but when I drop this into any file in my project:

```
[@decco]
type bobby =
  | A
  | B
  | C;
```

I get the following error message:

```
  We've found a bug for you!
  (No file name)

  The constructor Js.Json.JSONString belongs to the variant type
    Js.Json.tagged_t = Js_json.tagged_t
  but a constructor was expected belonging to the variant type option
```

Decco version 0.1.0, bs-platform version 5.2.1.

Thank you!

**Comment by baransu** [comment]
Experiencing the same issue 🙁

**Comment by ryb73** [comment]
I'm not able to reproduce this. Can you provide your bsconfig?

**Comment by baransu** [comment]
Repo: https://github.com/baransu/ppx_decco-variant-error-repo
Code: 
```reason
[@decco]
type status =
  | Active
  | Inactive(float);

[@decco]
type user = {
  name: string,
  status,
};
```

**Comment by ryb73** [comment]
Thanks for the repo. It looks like it's caused by opening `Belt`. I'll take a closer look later

**Comment by ryb73** [comment]
TIL OCaml's array indexing syntax is just syntactic sugar for `Array.get`. In other words, `var[0]` is equivalent to `Array.get(var, 0)`

This is the source of your error: OCaml's native `Array.get` returns the actual value directly (throwing an error if the index is out of bounds), while `Belt.Array.get` returns an option. When you open Belt, it changes the behavior of `var[0]`.

See: https://stackoverflow.com/a/46394530/1953740

I guess I can fix it by avoiding that array syntax and just using Belt.Array directly.

**Comment by ryb73** [comment]
Deployed a fix in version v0.2.2 for BS 6 and v0.1.1 for BS 5.

**Comment by baransu** [comment]
Thank you a lot!
